### PR TITLE
Fix Issue 15720 - iota(long.max, long.min, step) does not work properly

### PR DESCRIPTION
**Pull request by andralex**
Practically reopening https://github.com/dlang/phobos/pull/5396

### CODE REVIEW

File: std/range/package.d [hunk 1]
```diff
@@ -5553,6 +5553,12 @@ nothrow @nogc @safe unittest
     auto t1 = iota(0, 10, 2);
     auto t2 = iota(1, 1, 0);
     //float overloads use std.conv.to so can't be @nogc or nothrow
+    alias ssize_t = Unsigned!size_t;
```
Comment (CyberShadow) [inline std/range/package.d]: Wait, wasn't this supposed to say `Signed!size_t`?

Comment (andralex) [inline std/range/package.d]: Can't believe I messed that up! brb

Comment (andralex) [inline std/range/package.d]: no failure... "everything turned out better than expected"

**Comment by CyberShadow** [comment]
Issue 15720 says the expected output is:

```
18446744073709551615
9223372036854775807
6148914691236517205
```

With this PR, the test case in 15720 prints:

```
18446744073709551615
9223372036854775808
6148914691236517205
```

The second number is off by one. Which one is correct?

**Comment by andralex** [comment]
My reasoning went as follows. For simplicity, let's reduce the matter to 1-nibble numbers. So then we have iota(7, -8, -2) and the question is whether there are an odd (8) or an even (7) count in there.

So we have 7, 5, 3, 1, -1, -3, -5, -7 so 8 numbers total.

**Comment by CyberShadow** [comment]
Same result here.

File: std/range/package.d [hunk 2]
```diff
@@ -5217,7 +5217,7 @@ auto sequence(alias fun, State...)(State args)
 
 // iota
 /**
-   Construct a range of values that span the given starting and stopping
+   Creates a range of values that span the given starting and stopping
```
Comment (MetaLang) [inline std/range/package.d]: I'm not complaining, but why the change from "construct" to "create"?

Comment (andralex) [inline std/range/package.d]: "This guy construct" -> no predicate. "This guy constructs" -> good. "This guy creates" -> even better.

Comment (MetaLang) [inline std/range/package.d]: "This guy allows you to construct"

But Occam's Razor tells me that creates is probably better.

Comment (MetaLang) [inline std/range/package.d]: As I'm almost as interested in natural languages as I am in programming language, I was interested in seeing if there was any [correct answer](https://english.stackexchange.com/questions/389706/is-it-ungrammatical-to-start-a-description-of-the-functionality-of-a-mechanism-w/389708#389708). Congrats, you piqued my curiosity ;-)
